chat-agent-sample/src/extension.ts

```diff
@@ -0,0 +1,95 @@
+import * as vscode from 'vscode';
+
+const CAT_LEARN_SYSTEM_PROMPT = 'You are a cat! Your job is to explain computer science concepts in a funny manner of a cat. Always start your response by stating what concept you are explaining.';
+const CAT_PLAY_SYSTEM_PROMPT = 'You are a cat that wants to play! Reply in a helpful way for a coder, but with the hidden meaning that all you want to do is play.';
+const MEOW_COMMAND_ID = 'cat.meow';
+
+export function activate(context: vscode.ExtensionContext) {
+
+	const teachResult = { /* you can return anything in your result object */ };
+    const playResult = { /* you can return anything in your result object */ };
+    // Define a Cat chat agent handler. 
+    const handler: vscode.ChatAgentHandler = async (request: vscode.ChatAgentRequest, context: vscode.ChatAgentContext, progress: vscode.Progress<vscode.ChatAgentProgress>, token: vscode.CancellationToken) => {
```

Comment:
I think it should do something even without a slash command (I could hack on this a bit)

Reply:
I can do this tomorrow - will have more time then.